pkg/cmd/run/watch/watch.go

```diff
@@ -0,0 +1,228 @@
+		if err != nil {
+			return err
+		}
+		// TODO silly stopgap until dust settles and PromptForRun can just return a run
+		for _, r := range runs {
+			if fmt.Sprintf("%d", r.ID) == runID {
+				run = &r
+				break
+			}
+		}
+	} else {
+		run, err = shared.GetRun(client, repo, runID)
+		if err != nil {
+			return fmt.Errorf("failed to get run: %w", err)
+		}
+	}
+
+	if run.Status == shared.Completed {
+		return nil
+	}
+
+	prNumber := ""
+	number, err := shared.PullRequestForRun(client, repo, *run)
+	if err == nil {
+		prNumber = fmt.Sprintf(" #%d", number)
+	}
+
+	if runtime.GOOS == "windows" {
+		opts.IO.EnableVirtualTerminalProcessing()
+	}
+	// clear entire screen
+	fmt.Fprintf(opts.IO.Out, "\x1b[2J")
+
+	annotationCache := map[int][]shared.Annotation{}
+
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
+	if err != nil {
+		return fmt.Errorf("could not parse interval: %w", err)
+	}
+
+	for run.Status != shared.Completed {
+		run, err = renderRun(*opts, client, repo, run, prNumber, annotationCache)
+		if err != nil {
+			return err
+		}
+		time.Sleep(duration)
+	}
+
+	if opts.ExitStatus && run.Conclusion != shared.Success {
+		return cmdutil.SilentError
+	}
+
+	return nil
+}
+
+func renderRun(opts WatchOptions, client *api.Client, repo ghrepo.Interface, run *shared.Run, prNumber string, annotationCache map[int][]shared.Annotation) (*shared.Run, error) {
+	out := opts.IO.Out
+	cs := opts.IO.ColorScheme()
+
+	var err error
+
+	run, err = shared.GetRun(client, repo, fmt.Sprintf("%d", run.ID))
+	if err != nil {
+		return run, fmt.Errorf("failed to get run: %w", err)
+	}
+
+	ago := opts.Now().Sub(run.CreatedAt)
+
+	jobs, err := shared.GetJobs(client, repo, *run)
+	if err != nil {
+		return run, fmt.Errorf("failed to get jobs: %w", err)
+	}
+
+	var annotations []shared.Annotation
+
+	var annotationErr error
+	var as []shared.Annotation
+	for _, job := range jobs {
+		if as, ok := annotationCache[job.ID]; ok {
+			annotations = as
+			continue
+		}
+
+		as, annotationErr = shared.GetAnnotations(client, repo, job)
+		if annotationErr != nil {
+			break
+		}
+		annotations = append(annotations, as...)
+
+		if job.Status != shared.InProgress {
+			annotationCache[job.ID] = annotations
+		}
+	}
+
+	if annotationErr != nil {
+		return run, fmt.Errorf("failed to get annotations: %w", annotationErr)
+	}
+
+	if runtime.GOOS == "windows" {
+		// Just clear whole screen; I wasn't able to get the nicer cursor movement thing working
+		fmt.Fprintf(opts.IO.Out, "\x1b[2J")
+	} else {
+		// Move cursor to 0,0
+		fmt.Fprint(opts.IO.Out, "\x1b[0;0H")
+		// Clear from cursor to bottom of screen
+		fmt.Fprint(opts.IO.Out, "\x1b[J")
+	}
+
+	fmt.Fprintln(out, cs.Boldf("Refreshing run status every %d seconds. Press Ctrl+C to quit.", opts.Interval))
+	fmt.Fprintln(out)
+	fmt.Fprintln(out, shared.RenderRunHeader(cs, *run, utils.FuzzyAgo(ago), prNumber))
+	fmt.Fprintln(out)
+
+	if len(jobs) == 0 && run.Conclusion == shared.Failure {
```

Comment:
If there are no jobs in a run should we be returning here regardless of the status?